File: server/EvalutionAlgoritmCpp/EvalutionAlgoritmCpp/EvulationAlgoritmCpp/AddClassToSchedule.cpp
#include "AddClassToSchedule.h"
#include <iostream>

void AddClassToSchedule(individ& i_schedule, const clas& clas, const int& day_week, const int& number_pair, const int& pair_type, const int& id_audience)
{
    for (assigned_group ag : clas.assigned_groups)
    {
        vector<schedule> temp = i_schedule.scheduleForGroups[ag.id_group];

        temp.push_back(schedule(0, number_pair, day_week, pair_type, ag.id, id_audience, clas.id));

        i_schedule.scheduleForGroups[ag.id_group] = temp;
    }
    for (assigned_teacher at : clas.assigned_teachers)
    {
        vector<schedule> temp = i_schedule.scheduleForTeachers[at.id_teacher];

        temp.push_back(schedule(0, number_pair, day_week, pair_type, -1, id_audience, clas.id));

        i_schedule.scheduleForTeachers[at.id_teacher] = temp;
    }
    vector<schedule> temp = i_schedule.scheduleForAudiences[id_audience];
    temp.push_back(schedule(0, number_pair, day_week, pair_type, -1, id_audience, clas.id));
    i_schedule.scheduleForAudiences[id_audience] = temp;   
}

File: server/Algorithms/TabuSearchAlgorithm/AddClassToSchedule.js
export default function AddClassToSchedule(
  schedule,
  clas,
  day_week,
  number_pair,
  pair_type,
  id_audience,
  id_schedule,
  tabuList
) {
  // Для каждой группы добавление расписания
  // Для каждой группы добавление расписания
  clas.assigned_groups.forEach((ag) => {
    let temp = schedule.scheduleForGroups.get(ag.id_group);
    if (!temp) {
      temp = [];
    }
    temp.push({
      id: id_schedule,
      number_pair,
      day_week,
      pair_type,
      id_audience,
      clas,
    });
    schedule.scheduleForGroups.set(ag.id_group, temp);
  });
  // Для каждого учителя добавление расписания
  clas.assigned_teachers.forEach((at) => {
    let temp = schedule.scheduleForTeachers.get(at.id_teacher);
    if (!temp) {
      temp = [];
    }
    temp.push({
      id: id_schedule,
      number_pair,
      day_week,
      pair_type,
      id_audience,
      clas,
    });
    schedule.scheduleForTeachers.set(at.id_teacher, temp);
  });
  // Для  аудитории добавление расписания
  let temp = schedule.scheduleForAudiences.get(id_audience);
  if (!temp) {
    temp = [];
  }
  temp.push({
    id: id_schedule,
    number_pair,
    day_week,
    pair_type,
    id_audience,
    clas,
  });
  schedule.scheduleForAudiences.set(id_audience, temp);

  temp = tabuList.get(+clas.id);
  if (!temp) temp = new Array();
  temp.push(JSON.stringify({ day_week, number_pair, pair_type }));

  tabuList.set(clas.id, temp);
}